### DISCUSSION TITLE
how to bind v-model:value in render function h()?

### DISCUSSION THREAD

**Discussion by summerset-zz** [tusen-ai/naive-ui#3947] [Q&A] [answered]
I'm designing a dialog with NInput, and I want the positiveButton to be disabled when the input value has an empty string filled.

current code is like this:
```typescript
const inputRef = ref('');
dialog.info({
    content: () =>
        h('div', {}, [
            `sometext`, 
            // it's about this line
            h(NInput, { modelValue: inputRef }),
        ]),
    positiveText: 'confirm',
    positiveButtonProps: {
        disabled: inputRef.value.length == 0,
    },
    negativeText: 'cancel',
    onPositiveClick: () => {
        // promise handler
    },
});
```
It now performed that the positiveButton never turned enable, guess that the input model not working.

I've already tried ` h(NInput, { modelValue: inputRef })`,` h(NInput, { vModelValue: inputRef })` or exposing inputRef value to h() but not working.

maybe `v-model:value` is some kind of directive?

**Comment by XieZongChen** [comment, marked as answer]
In vue, `v-model:value` is the syntactic sugar of `value` and `on-update:value`. So when you use components through `h`, you can:
```
h(NInput, {
            value: row.name,
            onUpdateValue (v) {
              data.value[index].name = v
            }
})
```

**Reply by XieZongChen** [reply]
See https://www.naiveui.com/zh-CN/os-theme/components/data-table#simple-editable.vue
